pkgs/misc/cups/drivers/cnijfilter_3_80/default.nix

```diff
@@ -0,0 +1,159 @@
+
+    cd ../backendnet
+    ./autogen.sh --prefix=$out --enable-libpath=$out/lib/bjlib --enable-progpath=$out/bin
+
+    cd ../cnijfilter
+    ./autogen.sh --prefix=$out --enable-libpath=$out/lib/bjlib --enable-binpath=$out/bin
+
+    cd ../maintenance
+    ./autogen.sh --prefix=$out --datadir=$out/share --enable-libpath=$out/lib/bjlib
+
+    cd ../lgmon
+    ./autogen.sh --prefix=$out --enable-progpath=$out/bin
+
+    cd ../cngpijmon
+    ./autogen.sh --prefix=$out  --enable-progpath=$out/bin --datadir=$out/share
+
+    cd ../cngpijmon/cnijnpr
+    ./autogen.sh --prefix=$out --enable-libpath=$out/lib/bjlib
+  '';
+
+  preInstall = ''
+    mkdir -p $out/bin $out/lib/cups/filter $out/share/cups/model;
+  '';
+
+  postInstall = ''
+    set -o xtrace
+    for pr in ${models}; do
+      cd ppd;
+      ./autogen.sh --prefix=$out --program-suffix=$pr
+      make clean;
+      make;
+      make install;
+
+      cd ../lgmon
+      ./autogen.sh --prefix=$out --program-suffix=$pr --enable-progpath=$out/bin
+      make clean;
+      make;
+      make install;
+
+      cd ../cnijfilter;
+      ./autogen.sh --prefix=$out --program-suffix=$pr --enable-libpath=/var/lib/cups/path/lib/bjlib --enable-binpath=$out/bin;
+      make clean;
+      make;
+      make install;
+
+      cd ../maintenance
+      ./autogen.sh --prefix=$out --program-suffix=$pr --datadir=$out/share --enable-libpath=/var/lib/cups/path/lib/bjlib
+      make clean;
+      make;
+      make install;
+
+      cd ../cngpijmon
+      ./autogen.sh --prefix=$out --program-suffix=$pr --enable-progpath=$out/bin --datadir=$out/share
+      make clean;
+      make;
+      make install;
+
+      cd ..;
+    done;
+
+    mkdir -p $out/lib/bjlib;
+    for pr_id in ${numbers}; do
+      install -c -m 755 $pr_id/database/* $out/lib/bjlib;
+      install -c -s -m 755 $pr_id/libs_bin${arch}/*.so.* $out/lib;
+    done;
+
+    pushd $out/lib;
+    for so_file in *.so.*; do
+      ln -s $so_file ''${so_file/.so.*/}.so;
+      patchelf --set-rpath $out/lib $so_file;
+    done;
+    popd;
+  '';
+
+  /* the tarball includes some pre-built shared libraries.  we run
+     'patchelf --set-rpath' on them just a few lines above, so that
+     they can find each other.  but that's not quite enough.  some of
+     those libraries load each other in non-standard ways -- they
+     don't list each other in the DT_NEEDED section.  so, if the
+     standard 'patchelf --shrink-rpath' (from
+     pkgs/development/tools/misc/patchelf/setup-hook.sh) is run on
+     them, it undoes the --set-rpath.  this prevents that. */
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description = "Canon InkJet printer drivers for the ${models} series.";
+    homepage = https://www.canon.co.uk/support/consumer_products/products/printers/inkjet/pixma_ip_series/pixma_ip7250.html?type=drivers&language=&os=linux;
+    license = licenses.unfree;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ turion ];
```

Comment:
I can't help with the actual implementation or testing, but I can help you make ofborg happy: you need to add yourself as a maintainer for this to be valid. See [`maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) for more information, and my maintainers commit for an example: https://github.com/NixOS/nixpkgs/commit/ecf79a07e1cf1d895b058d33def554ae7c35283d.

Reply:
Thanks!